mobile_verifier/tests/rewarder_mappers.rs

```diff
@@ -0,0 +1,151 @@
+                subscriber_rewards[0].subscriber_id
+            );
+            assert_eq!(5_464_480_874_316, subscriber_rewards[0].discovery_location_amount);
+
+            assert_eq!(
+                SUBSCRIBER_2.to_string().encode_to_vec(),
+                subscriber_rewards[1].subscriber_id
+            );
+            assert_eq!(5_464_480_874_316, subscriber_rewards[2].discovery_location_amount);
+
+            assert_eq!(
+                SUBSCRIBER_3.to_string().encode_to_vec(),
+                subscriber_rewards[2].subscriber_id
+            );
+            assert_eq!(5_464_480_874_316, subscriber_rewards[2].discovery_location_amount);
+
+            // confirm our unallocated amount
+            assert_eq!(
+                UnallocatedRewardType::Mapper as i32,
+                unallocated_reward.reward_type
+            );
+            assert_eq!(2, unallocated_reward.amount);
+
+            // confirm the total rewards allocated matches expectations
+            let expected_sum = reward_shares::get_scheduled_tokens_for_mappers(epoch.end - epoch.start)
+                .to_u64()
+                .unwrap();
+            let subscriber_sum = subscriber_rewards[0].discovery_location_amount
+                    + subscriber_rewards[1].discovery_location_amount
+                    + subscriber_rewards[2].discovery_location_amount
+                    + unallocated_reward.amount;
+            assert_eq!(
+                expected_sum,
+                subscriber_sum
+            );
+
+            // confirm the rewarded percentage amount matches expectations
+            let daily_total = reward_shares::get_total_scheduled_tokens(epoch.end - epoch.start);
+            let percent = (Decimal::from(subscriber_sum) / daily_total).round_dp_with_strategy(2, RoundingStrategy::MidpointNearestEven);
+            assert_eq!(percent, dec!(0.2));
+        }
+    );
+    Ok(())
+}
+
+async fn receive_expected_rewards(
+    mobile_rewards: &mut MockFileSinkReceiver,
+) -> anyhow::Result<(Vec<SubscriberReward>, UnallocatedReward)> {
+    // get the filestore outputs from rewards run
+    // we will have 3 radio rewards, 1 wifi radio and 2 cbrs radios
+    let subscriber_reward1 = mobile_rewards.receive_subscriber_reward().await;
+    let subscriber_reward2 = mobile_rewards.receive_subscriber_reward().await;
+    let subscriber_reward3 = mobile_rewards.receive_subscriber_reward().await;
+    let subscriber_rewards = vec![subscriber_reward1, subscriber_reward2, subscriber_reward3];
+
+    // expect one unallocated reward
+    let unallocated_reward = mobile_rewards.receive_unallocated_reward().await;
+
+    // should be no further msgs
+    mobile_rewards.assert_no_messages();
+
+    Ok((subscriber_rewards, unallocated_reward))
+}
+
+async fn seed_mapping_data(
+    ts: DateTime<Utc>,
+    txn: &mut Transaction<'_, Postgres>,
+) -> anyhow::Result<()> {
+    // subscriber 1 has two mapping event during epoch
```

Comment:
might clarify that these should not be mapping events but qualifying mapping criteria reports (the boolean value that verifies rewardable activity was submitted to the service provider) and that this test ensures that the verifier is properly ignoring multiple reports for the same subscriber within the same epoch

Reply:
addressed here: 6dce19e2e546c3fd3316b7fdde8e446d1bcfaf85 and comments in the test itself outline that subscribers receive same share irrespective of number of reports/events